zulipterminal/model.py

```diff
@@ -1123,7 +1123,10 @@ def _update_rendered_view(self, msg_id: int) -> None:
                 # narrow.
                 if (len(self.narrow) == 2
                         and msg_box.message['subject'] != self.narrow[1][1]):
+                    msg_pos = view.message_view.log.index(msg_w)
                     view.message_view.log.remove(msg_w)
+                    prev_msg_sender = msg_box.last_message['sender_email']
```

Comment:
The tests seem to be failing on this line. You need to specify a mock value of `sender_email` in the `test_model.py` module.